awscli/customizations/codeartifact/login.py

```diff
@@ -303,6 +309,12 @@ def login(self, dry_run=False):
         )
         self._run_commands('npm', commands, dry_run)
 
+    def _run_command(self, tool, command):
```

Comment:
We can simplify the logic here since `ignore_errors` is a boolean. Then, we do not need an if statement doing the work for us. We can do something like this:

```
ignore_errors = any('always-auth' in arg for arg in command)
super()._run_command(tool, command, ignore_errors=ignore_errors)
```

Separately, is it possible for us to be more specific in checking where `always-auth` occurs? What scenarios could it be in different (or multiple) positions?

Reply:
I can make the first change. For second question, we only have one command in npm which has "always-auth" (particular keyword) setting and we are only checking if the command we are going to run has it, i am not sure if i understand the question "what scenarios could it be in different positions?"